gateway/src/main/java/org/georchestra/gateway/filter/headers/RemoveHeadersGatewayFilterFactory.java

```diff
@@ -78,7 +78,7 @@ public GatewayFilter apply(RegExConfig regexConfig) {
         return (exchange, chain) -> {
             final RegExConfig config = regexConfig;// == null ? DEFAULT_SECURITY_HEADERS_CONFIG : regexConfig;
             HttpHeaders incoming = exchange.getRequest().getHeaders();
-            if (config.anyMatches(incoming)) {
+            if (config.anyMatches(incoming) && !incoming.containsKey("sec-mellon-name-id")) {
```

Comment:
I think the name sounds too specific, we may want something more "generic". 

Also in the previous security-proxy we were relying on the host (name / IP address) from where the request was coming (see https://github.com/georchestra/georchestra/blob/master/security-proxy/src/main/java/org/georchestra/security/ProxyTrustAnotherProxy.java#L62-L85), instead of a header, which sounds safer:
what if the user sends a "sec-mellon-name-id" header in ones http request ? I'd expect he/she will be trusted. It sounds safer to check the request is coming from apache.

Reply:
this has been fixed in the following commits, but still require some documentation on how to configure the filter (`sec-georchestra-preauthenticated: true` header, and so on...)